File: javascript/01_authenticate.js
'use strict';
/**
 * Step 1: Verify authentication with Salesforce.
 *
 * Run with:
 *   node 01_authenticate.js
 */

const { getConnection } = require('./auth');

async function main() {
    console.log('Connecting to Salesforce...');
    const conn = await getConnection();

    // conn.instanceUrl contains the org's base URL
    // conn.userInfo contains the authenticated user's details
    const userInfo = conn.userInfo;
    console.log(`Connected to Salesforce org: ${userInfo.organizationId}`);
    console.log(`Instance URL: ${conn.instanceUrl}`);
    console.log(`Authenticated as user: ${userInfo.id}`);
    console.log('Authentication successful.');
}

main().catch(err => {
    console.error('Authentication failed:', err.message);
    process.exit(1);
});

File: python/01_authenticate.py
"""Step 1: Verify authentication with Salesforce.

Run with:
    uv run python 01_authenticate.py
"""

from auth import get_salesforce_connection


def main():
    print("Connecting to Salesforce...")
    sf = get_salesforce_connection()

    # sf.sf_instance contains the org's instance URL (e.g., yourorg.my.salesforce.com)
    # sf.sf_org_id contains the 18-character org ID
    print(f"Connected to Salesforce org: {sf.sf_org_id}")
    print(f"Instance URL: https://{sf.sf_instance}")
    print("Authentication successful.")


if __name__ == "__main__":
    main()
